stackit/internal/features/beta_test.go

```diff
@@ -0,0 +1,285 @@
+			data:        &core.ProviderData{},
+			expected:    false,
+		},
+		{
+			description: "Feature flag not set, Env var is true",
+			data:        &core.ProviderData{},
+			envSet:      true,
+			envValue:    "true",
+			expected:    true,
+		},
+		{
+			description: "Feature flag not set, Env var is false",
+			data:        &core.ProviderData{},
+			envSet:      true,
+			envValue:    "false",
+			expected:    false,
+		},
+		{
+			description: "Feature flag not set, Env var is empty",
+			data:        &core.ProviderData{},
+			envSet:      true,
+			envValue:    "",
+			expectWarn:  true,
+			expected:    false,
+		},
+		{
+			description: "Feature flag not set, Env var is gibberish",
+			data:        &core.ProviderData{},
+			envSet:      true,
+			envValue:    "gibberish",
+			expectWarn:  true,
+			expected:    false,
+		},
+		{
+			description: "Feature flag enabled, Env var is true",
+			data: &core.ProviderData{
+				EnableBetaResources: true,
+			},
+			envSet:   true,
+			envValue: "true",
+			expected: true,
+		},
+		{
+			description: "Feature flag enabled, Env var is false",
+			data: &core.ProviderData{
+				EnableBetaResources: true,
+			},
+			envSet:   true,
+			envValue: "false",
+			expected: false,
+		},
+		{
+			description: "Feature flag enabled, Env var is empty",
+			data: &core.ProviderData{
+				EnableBetaResources: true,
+			},
+			envSet:     true,
+			envValue:   "",
+			expectWarn: true,
+			expected:   true,
+		},
+		{
+			description: "Feature flag enabled, Env var is gibberish",
+			data: &core.ProviderData{
+				EnableBetaResources: true,
+			},
+			envSet:     true,
+			envValue:   "gibberish",
+			expectWarn: true,
+			expected:   true,
+		},
+		{
+			description: "Feature flag disabled, Env var is true",
+			data: &core.ProviderData{
+				EnableBetaResources: false,
+			},
+			envSet:   true,
+			envValue: "true",
+			expected: true,
+		},
+		{
+			description: "Feature flag disabled, Env var is false",
+			data: &core.ProviderData{
+				EnableBetaResources: false,
+			},
+			envSet:   true,
+			envValue: "false",
+			expected: false,
+		},
+		{
+			description: "Feature flag disabled, Env var is empty",
+			data: &core.ProviderData{
+				EnableBetaResources: false,
+			},
+			envSet:     true,
+			envValue:   "",
+			expectWarn: true,
+			expected:   false,
+		},
+		{
+			description: "Feature flag disabled, Env var is gibberish",
+			data: &core.ProviderData{
+				EnableBetaResources: false,
+			},
+			envSet:     true,
+			envValue:   "gibberish",
+			expectWarn: true,
+			expected:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			var err error
+			var lastEnv string
+			var envWasSet bool
+			if tt.envSet {
+				lastEnv, envWasSet = os.LookupEnv("STACKIT_TF_ENABLE_BETA_RESOURCES")
+				err = os.Setenv("STACKIT_TF_ENABLE_BETA_RESOURCES", tt.envValue)
+				if err != nil {
+					t.Fatalf("Error setting environment variable: %s", err)
+				}
+			}
```

Comment:
:+1: for the complete test!

Reply:
One suggestion: I think you can use [`t.Setenv`](https://pkg.go.dev/testing#B.Setenv), which sets the environment variable for the duration of the current test only

Reply:
Thanks, that actually simplifies the test a lot